### PR TITLE
[11.8] Add resource events API endpoints

### PR DESCRIPTION
**Pull request by joshuachp**
Hi, like #666 that implements the Resource event API, but I needed all 3 types of events.

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -361,19 +431,6 @@ public function authenticate(string $token, string $authMethod, string $sudo = n
         $this->getHttpClientBuilder()->addPlugin(new Authentication($authMethod, $token, $sudo));
     }
 
-    /**
```
Comment (GrahamCampbell) [inline src/Client.php]: Please revert this change.

File: src/Client.php [hunk 2]
```diff
@@ -403,14 +460,4 @@ public function getStreamFactory(): StreamFactoryInterface
     {
         return $this->getHttpClientBuilder()->getStreamFactory();
     }
-
```
Comment (GrahamCampbell) [inline src/Client.php]: Please revert this change.

File: src/Client.php [hunk 3]
```diff
@@ -116,14 +121,39 @@ public function __construct(Builder $httpClientBuilder = null)
 
         $builder->addPlugin(new ExceptionThrower());
         $builder->addPlugin(new HistoryPlugin($this->responseHistory));
-        $builder->addPlugin(new HeaderDefaultsPlugin([
-            'User-Agent' => self::USER_AGENT,
-        ]));
+        $builder->addPlugin(
```
Comment (GrahamCampbell) [inline src/Client.php]: Please revert this change.

File: .gitignore [hunk 4]
```diff
@@ -3,3 +3,6 @@ composer.lock
 phpstan.neon
 phpunit.xml
 vendor
+
+composer.phar
```
Comment (GrahamCampbell) [inline .gitignore]: Please revert this change.

File: .gitignore [hunk 5]
```diff
@@ -3,3 +3,6 @@ composer.lock
...
+.idea/
```
Comment (GrahamCampbell) [inline .gitignore]: Please revert this change.

File: .gitignore [hunk 6]
```diff
@@ -3,3 +3,6 @@ composer.lock
 phpstan.neon
 phpunit.xml
 vendor
+
```
Comment (GrahamCampbell) [inline .gitignore]: Please revert this change.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]
Thanks for the PR. I've left some inline changes.

File: src/Client.php (hunk 1 above)
Comment (joshuachp) [reply src/Client.php]: Reverted in d18e1d10fcd4f2768afcef81c011bcc29ce0c1e1

File: src/Client.php (hunk 2 above)
Comment (joshuachp) [reply src/Client.php]: Reverted in d18e1d10fcd4f2768afcef81c011bcc29ce0c1e1

File: src/Client.php (hunk 3 above)
Comment (joshuachp) [reply src/Client.php]: Reverted in d18e1d10fcd4f2768afcef81c011bcc29ce0c1e1

File: .gitignore (hunk 4 above)
Comment (joshuachp) [reply .gitignore]: Reverted in d18e1d10fcd4f2768afcef81c011bcc29ce0c1e1

File: .gitignore (hunk 5 above)
Comment (joshuachp) [reply .gitignore]: Reverted in d18e1d10fcd4f2768afcef81c011bcc29ce0c1e1

File: .gitignore (hunk 6 above)
Comment (joshuachp) [reply .gitignore]: Reverted in d18e1d10fcd4f2768afcef81c011bcc29ce0c1e1

**Comment by joshuachp** [comment]
Reverted changes in d18e1d10fcd4f2768afcef81c011bcc29ce0c1e1

File: tests/Api/ResourceIterationEventsTest.php [hunk 7]
```diff
@@ -0,0 +1,131 @@
+<?php
```
Comment (GrahamCampbell) [inline tests/Api/ResourceIterationEventsTest.php:1]: Please add the copyright header to all files.

```php
<?php

declare(strict_types=1);

/*
 * This file is part of the Gitlab API library.
 *
 * (c) Matt Humphrey <matth@windsor-telecom.co.uk>
 * (c) Graham Campbell <hello@gjcampbell.co.uk>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

namespace Gitlab
```

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: tests/Api/ResourceIterationEventsTest.php (hunk 7 above)
Comment (joshuachp) [reply tests/Api/ResourceIterationEventsTest.php:1]: Added in 026697f64a51f75bb4ff1dc66e87b1ee06b366ce

**Comment by GrahamCampbell** [comment]
Excellent. Thanks for working on this! 🎉

**Review by GrahamCampbell — APPROVED** [review]
